### ISSUE TITLE
Unable to access response headers when using axios 1.x with jest

### ISSUE THREAD

**Issue by johannes-vogel** [axios/axios#5017]
#### Describe the bug
When using axios to send requests within jest tests, it's not possible to see the response.headers anymore.
This was working fine with axios version 0.x so far.

#### To Reproduce

I've prepared a small example to show that it it is not working as expected.
```js
test('xxx', (done) => {
  const express = require('express')
  const app = express()
  
  app.get('/test', (req, res) => {
    console.log('request received')
    res.set('x-header', 'value')
    res.send('ok')
  })
  
  const server = app.listen(4004, async () => {
    const a = require('axios')
    const axios = a.create({
      headers: { 'Content-Type': 'application/json' },
      baseURL: 'http://localhost:4004'
    })
    let response = await axios.get('/test')
    console.log(response.headers)
    server.close()
    done()
  })
})
```

On console, I can see the following log:
```log
  console.log
    AxiosHeaders { '[object Object]': undefined, [Symbol(defaults)]: null }
      at Server.log (/Users/j/Downloads/odata_call (1)../../../../../axios_issue/plain.test.js:18:13)
```

#### Expected behavior

I'd expect that the response headers are also accessible within the jest environment.

#### Solution proposal

I've narrowed it down that the response headers are normalized here:
https://github.com/axios/axios/blob/738fa63661a7a5d7e8bf604436bb3b91648e327b/lib/adapters/http.js#L427-L433
Within the setter of the AxiosHeaders there's a strict check whether the headers are an object:
https://github.com/axios/axios/blob/bdf493cf8b84eb3e3440e72d5725ba0f138e0451/lib/core/AxiosHeaders.js#L114

Somehow by jest magic, the `header.prototype` is not `Object.prototype` (even though it looks like it would be), hence the check returns false, even though it's an object.
In my naive opinion, the strict check is not required.

Another option would be to restore the previous behavior. With version 0.27.2, there was no normalization using AxiosHeaders:
https://github.com/axios/axios/blob/bc733fec78326609e751187c9d453cee9bf1993a/lib/adapters/http.js#L288-L294

#### Environment
 - axios version 1.0.0
 - express version 4.18.1 (probably unrelated)
 - jest version 29.1.2
 - node version 14.20.0

#### Additional context/Screenshots

I've also created a small sample repository to reproduce the problem even easier:
https://github.com/johannes-vogel/axios-jest-issue

- checkout the project
- npm i
- node no-jest.js → You'll observe it working without jest
- npx jest with-jest.test.js → You'll observe the mentioned log output.

**Comment by grenik** [comment]
Seems relevant to https://github.com/axios/axios/issues/5089 which shall be fixed in https://github.com/axios/axios/pull/5090

**Comment by johannes-vogel** [comment]
It's fixed already with version 1.1.2.
@risa and @jasonsaayman Thanks for providing a quick solution 👍
